subsys/bluetooth/controller/ll_sw/ctrl_internal.h

```diff
@@ -128,8 +130,8 @@ struct connection {
 		} slave;
 	};
 
-	u8_t  llcp_req;
-	u8_t  llcp_ack;
+	atomic_var_t  llcp_req;
+	atomic_var_t  llcp_ack;
```

Comment:
Since this concept is used a few places, it would make sense to collapse `req` and `ack` into 1 struct.
And only pass in this struct to the macros/inline functions, rather than referring to `req` and `ack` all over.
This makes sense because `req` and `ack` belong together and together implement a kind of software transactional / lock-free synchronization primitive.

Reply:
Good point @mped-oticon. According to @cvinayak, a single atomic variable may suffice. If you could clarify this approach with me at your convenience that would be appreciated.